### PR TITLE
added decorators for EventGrid.

### PR DESCRIPTION
**Pull request by pdthummar**

### CODE REVIEW

File: azure/functions/decorators/function_app.py [hunk 1]
```diff
@@ -1373,3 +1375,94 @@ def decorator():
...
+                        data_type=parse_singular_param_to_enum(data_type,
+                                                               DataType),
+                        **kwargs))
+                return fb
+
+            return decorator()
+
+        return wrap
+
+    def write_event_grid(self,
```
Comment (gavin-aguiar) [inline azure/functions/decorators/function_app.py:1420]: Did we finalize of this name? write_event_grid_message would be more appropriate here IMO. Thoughts? @YunchuWang @vrdmr @shreyabatra4

Comment (gavin-aguiar) [inline azure/functions/decorators/function_app.py:1420]: Did we finalize of this name? write_event_grid_message would be more appropriate here IMO. Thoughts? @YunchuWang @vrdmr @shreyabatra4

File: azure/functions/decorators/function_app.py [hunk 2]
```diff
@@ -1373,3 +1375,94 @@ def decorator():
...
+        to the :class:`FunctionBuilder` object
+        for building :class:`Function` object used in worker function
+        indexing model. This is equivalent to defining write_event_grid
+        in the function.json which enables function to
+        write events to a custom topic.
+        All optional fields will be given default value by function host when
+        they are parsed by function host.
+
+        Ref:
+        https://docs.microsoft.com/en-us/azure/azure-functions/functions-bindings-event-grid-trigger
```
Comment (gavin-aguiar) [inline azure/functions/decorators/function_app.py:1440]: Change to aka.ms link here too

Comment (gavin-aguiar) [inline azure/functions/decorators/function_app.py:1440]: Change to aka.ms link here too

File: azure/functions/decorators/function_app.py [hunk 3]
```diff
@@ -1373,3 +1375,94 @@ def decorator():
...
+        to the :class:`FunctionBuilder` object
+        for building :class:`Function` object used in worker function
+        indexing model. This is equivalent to defining event grid trigger
+        in the function.json which enables function to be triggered to
+        respond to an event sent to an event grid topic.
+        All optional fields will be given default value by function host when
+        they are parsed by function host.
+
+        Ref:
+        https://docs.microsoft.com/en-us/azure/azure-functions/functions-bindings-event-grid-trigger
```
[the remark this answers is not in the document]
Comment (shreyabatra4) [reply azure/functions/decorators/function_app.py:1396]: created - aka.ms/eventgridtrigger

[the remark this answers is not in the document]
Comment (shreyabatra4) [reply azure/functions/decorators/function_app.py:1396]: created - aka.ms/eventgridtrigger

File: azure/functions/decorators/function_app.py (hunk 1 above)
Comment (shreyabatra4) [reply azure/functions/decorators/function_app.py:1420]: as write_event_hub_message is what we are doing for event hub, write_event_grid_message may be more aligned

File: azure/functions/decorators/function_app.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (pdthummar) [reply azure/functions/decorators/function_app.py]: updated as per recommendation.

File: azure/functions/decorators/function_app.py (hunk 1 above)
Comment (pdthummar) [reply azure/functions/decorators/function_app.py]: updated as per recommendation.

Comment (pdthummar) [reply azure/functions/decorators/function_app.py]: updated as per recommendation.

File: azure/functions/decorators/function_app.py (hunk 2 above)
Comment (pdthummar) [reply azure/functions/decorators/function_app.py]: updated as per recommendation.

Comment (pdthummar) [reply azure/functions/decorators/function_app.py]: updated as per recommendation.

**Review by gavin-aguiar — APPROVED** [review]

**Review by vrdmr — APPROVED** [review]

File: docs/ProgModelSpec.pyi [hunk 4]
```diff
@@ -822,3 +822,64 @@ class FunctionApp:
 
         pass
 
+    def event_grid_trigger_message(self,
```
Comment (YunchuWang) [inline docs/ProgModelSpec.pyi:825]: to be consistent with other decorators, can you change to event_grid_trigger?

File: azure/functions/decorators/function_app.py [hunk 5]
```diff
@@ -1373,3 +1375,92 @@ def decorator():
             return decorator()
 
         return wrap
+
+    def event_grid_trigger_message(self,
```
Comment (YunchuWang) [inline azure/functions/decorators/function_app.py:1379]: to be consistent with other decorators, can you change to event_grid_trigger?

File: azure/functions/decorators/function_app.py [hunk 6]
```diff
@@ -1373,3 +1375,92 @@ def decorator():
...
+                        data_type=parse_singular_param_to_enum(data_type,
+                                                               DataType),
+                        **kwargs))
+                return fb
+
+            return decorator()
+
+        return wrap
+
+    def write_event_grid_message(self,
```
Comment (YunchuWang) [inline azure/functions/decorators/function_app.py:1419]: ditto, write_event_grid to be consistent with others

File: azure/functions/decorators/function_app.py (hunk 5 above)
Comment (gavin-aguiar) [reply azure/functions/decorators/function_app.py:1379]: good catch. I missed seeing this,

**Review by gavin-aguiar — APPROVED** [review]

**Review by YunchuWang — APPROVED** [review]

**Pull request merged**